storage/scanner_test.go

```diff
@@ -331,7 +332,11 @@ func TestScannerDisabled(t *testing.T) {
 	s.SetDisabled(true)
 	lastScannerCount := s.Count()
 	util.SucceedsSoon(t, func() error {
-		time.Sleep(2 * time.Millisecond)
+		// We want to make sure that a complete scanner loop has time to complete
+		// between every retry iteration, so twice the scan interval is a safe
+		// value. In addition, the internal timers used by the scanner can
+		// occasionally fire with delay, so we want to be extra patient (see #8709).
+		time.Sleep(2*scanInterval + 200*time.Millisecond)
```

Comment:
Hmm, didn't read code carefully enough. The main point of the comment is we want to avoid a hard wait of 200ms.

Maybe introduce a `waiting` bool which is true in `waitEnabled()` and verify with `SucceedsSoon` that that's set.